catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java

```diff
@@ -662,26 +662,42 @@ public Serializable parseDate(Serializable value) {
       return null;
     }
 
-    if (StringUtils.isEmpty((String) value)) {
+    String string = String.valueOf(value);
+
+    if (StringUtils.isBlank(string)) {
       return null;
     }
 
-    String string = String.valueOf(value);
+    if (StringUtils.isNumeric(string)) {
+      try {
+        return Instant.ofEpochMilli(Long.parseLong(string));
+      } catch (NumberFormatException ex) {
+        LOGGER.debug("Failed to create Epoch time from a numeric: {}", string, ex);
+        return null;
+      }
+    }
+
+    if (iso8601Z.matcher(string).matches()) {
+      return Instant.parse(string);
+    }
+
+    if (iso8601Offset.matcher(string).matches()) {
+      return OffsetDateTime.parse(string).toInstant();
+    }
 
     SimpleDateFormat dateFormat;
     if (boonDefault.matcher(string).matches()) {
       dateFormat = new SimpleDateFormat("EEE MMM d HH:mm:ss zzz yyyy");
-    } else if (iso8601.matcher(string).matches()) {
-      dateFormat = new SimpleDateFormat(ISO_8601_DATE_FORMAT);
     } else {
       dateFormat = new SimpleDateFormat();
     }
     dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
 
     try {
-      return dateFormat.parse(value.toString());
+      return dateFormat.parse(string).toInstant();
```

Comment:
In other places I've been using the `toInstant(ZoneOffset.UTC)`

Reply:
Same comment as above, except in this case on the Date object.  https://docs.oracle.com/javase/8/docs/api/java/util/Date.html